File: WondayWall.Android/app/src/main/kotlin/com/studiofreesia/wondaywall/models/UpdateSchedule.kt
package com.studiofreesia.wondaywall.models

import kotlinx.serialization.Serializable

// 壁紙の自動更新スケジュール（Windows版 UpdateSchedule と同名で揃える）
@Serializable
enum class UpdateSchedule {
    // 週1回（月曜 4:00）
    OnceAWeek,
    // 週2回（月曜・木曜 4:00）
    TwiceAWeek,
    // 週3回（月曜・水曜・金曜 4:00）
    ThreeTimesAWeek,
    // 1日1回（4:00）
    OnceADay,
    // 1日3回（朝4:00・昼12:00・晩18:00）
    ThreeTimesADay,
    ;

    // 画面表示用の日本語名
    fun displayName(): String = when (this) {
        OnceAWeek -> "週1回"
        TwiceAWeek -> "週2回"
        ThreeTimesAWeek -> "週3回"
        OnceADay -> "1日1回"
        ThreeTimesADay -> "1日3回"
    }
}

File: WondayWall/Models/UpdateSchedule.cs
namespace WondayWall.Models;

/// <summary>壁紙の自動更新スケジュール</summary>
public enum UpdateSchedule
{
    /// <summary>週1回（月曜 7:00）</summary>
    OnceAWeek = 0,
    /// <summary>週2回（月曜・木曜 7:00）</summary>
    TwiceAWeek = 1,
    /// <summary>週3回（月曜・水曜・金曜 7:00）</summary>
    ThreeTimesAWeek = 2,
    /// <summary>1日1回（0:00）</summary>
    OnceADay = 3,
    /// <summary>1日3回（朝7:00・昼12:00・晩18:00）</summary>
    ThreeTimesADay = 4,
}
